scheduler/src/cook/kubernetes/api.clj

```diff
@@ -682,10 +707,14 @@
         sandbox-volume-mount-fn #(make-volume-mount sandbox-volume sandbox-dir %)
         sandbox-volume-mount (sandbox-volume-mount-fn false)
         ; mesos-sandbox-volume-mount added for Mesos backward compatibility
-        mesos-sandbox-volume-mount (make-volume-mount sandbox-volume "/mnt/mesos/sandbox" false)]
+        mesos-sandbox-volume-mount (make-volume-mount sandbox-volume "/mnt/mesos/sandbox" false)
+        label-prefix (:add-job-label-to-pod-prefix (config/kubernetes))
+        make-shm-volume? (= "true" (get job-labels (str label-prefix "shared-memory")))
```

Comment:
Can we have an integration test for this?

Reply:
I don't have an opensource integration test I can run. How much do you want one?

Reply:
Maybe a simple test that submits a job with the label, and then writes to and reads from `/dev/shm`? And then a job that attempts to write to `/dev/shm` but without the label and fails?

Reply:
Doing it that way is is not so simple of a test --- We need to create a container and send it to GKE/K8s that contains an image that has a script that fails. I'd like to not open up the quagmire of image management.

I've tried a half dozen replicators, without success. The good news is that we can try various one-liners without needing to come up with an image --- and I have tried your idea, it failed.

I tried:

`cs submit -m 20 -c .001 --pool k8s-gamma "python3.8 -c 'import multiprocessing.shared_memory; multiprocessing.shared_memory.SharedMemory(None,True,1000000).unlink()'"`

`cs submit -m 20 -c .001 --pool k8s-gamma 'ls /dev/shm/'`
or even
`cs submit -m 20 -c .001 --pool k8s-gamma 'echo hello > /dev/shm/foo && rm -f /dev/shm/foo'`

But these jobs pass even with the label not set. I tried this, which (properly) fails, but this wouldn't succeed with the label set:

`echo hello > /dev/shmx/foo && rm -f /dev/shm/foo`

Whatever it is, these tests aren't exercising the pytorch codepaths that inspired the change. (Which I have validated are fixed by this change.)

Reply:
Got it, thanks for trying. Made a suggestion for a different test in the other comment thread; resolving this comment thread.